### DISCUSSION TITLE
How to Create TUI using new Command Annotations

### DISCUSSION THREAD

**Discussion by micsport13** [spring-projects/spring-shell#1133] [Q&A]
Basically, the title of this discussion.  I noticed in the docs that there isn't much information on how to combine creating terminal interfaces and having the style settings with command annotations.  Maybe I just don't understand Spring Shell that well but it seems kind of confusing to know where to place the Terminal and Styling settings in the spring shell project.  I just found out about the concept of TUI's in Spring shell today and the idea is very interesting to me since a large part of my app is interactive, which I am just doing through the standard output but the styling guide is very interesting.  Also, just as a side note, I love using Spring shell and I'm thankful for all the work you guys do on it.  Anyway, any clarity I can get here would be greatly appreciated.

**Comment by micsport13** [comment]
Also, how stable should I consider the TerminalUI classes and related classes?

**Comment by titimoby** [comment]
I found a lot of info in this series of blog articles: https://medium.com/agency04/developing-cli-application-with-spring-shell-part-1-807cd3a32461

The TerminalUI part seems really recent but I love the simple fact that it now exists ;)
